### PR TITLE
Feat cards

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
Various types of cards are added 

- Horizontal Card 
  - Image at center 
  - Image at top
- Vertical Card 
- Card with text overlay
- Text only card 
- Card with dismiss button

Preview: https://deploy-preview-12--hydrogen-ui.netlify.app/docs/card

### CODE REVIEW

File: components/card.css [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+.card i {
+  font-size: inherit;
+}
+
+.card-title {
+  margin-bottom: 0;
+}
+
+.card .discount-percent {
+  color: #006e00;
```
Comment (aryan-13) [inline components/card.css]: This color can be declared inside a variable!

File: components/card.css [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+.card-horizontal {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content: center;
+}
+
+.card-image {
+  width: 150px;
```
Comment (aryan-13) [inline components/card.css:32]: Pixels can be replaced with rems

File: components/card.css [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+.border-m {
+  border-radius: 10px;
+}
+
+.border-l {
+  border-radius: 15px;
+}
+
+.border-f {
```
Comment (aryan-13) [inline components/card.css]: The class name doesn't clearly tell the purpose of itself

**Review by aryan-13 — CHANGES_REQUESTED** [review]
Everything else looks good

File: components/card.css (hunk 1 above)
Comment (junaidshaikh-js) [reply components/card.css]: Noted!

File: components/card.css [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+/* TODO  */
+/* to be removed after adding utitlity classes  */
+
+/* border radius */
+
+.border-s {
+  border-radius: 5px;
+}
```
Comment (devananth) [inline components/card.css]: If you could name it as `.rounded` , `.rounded-md` , this would be meaningful.

File: components/card.css (hunk 1 above)
Comment (junaidshaikh-js) [reply components/card.css]: Fixed

File: components/card.css (hunk 2 above)
Comment (junaidshaikh-js) [reply components/card.css:32]: The card column needs to be of fixed width hence width in pixels is given.

**Review by devananth — COMMENTED** [review]
Great work !

File: components/card.css (hunk 3 above)
Comment (junaidshaikh-js) [reply components/card.css]: Let me change, thank you for pointing out.

File: components/card.css (hunk 4 above)
Comment (junaidshaikh-js) [reply components/card.css]: I was thinking to keep `border` and change the suffix.

**Pull request merged**